app/models/concerns/hyrax/with_depositor.rb

```diff
@@ -0,0 +1,11 @@
+module Hyrax::WithDepositor
+  # Adds metadata about the depositor to the asset and
+  # grants edit permissions to the +depositor+
+  # @param [String, #user_key] depositor
+  def apply_depositor_metadata(depositor)
```

Comment:
i think we should avoid putting this business logic into the model code. maybe the transaction step `SetUserAsDepositor` could be reused here?